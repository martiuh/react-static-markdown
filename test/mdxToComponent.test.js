const React = require('react');
const getFilesWithMdx = require('../getFilesWithMdx');
const mdxToComponent = require('../mdxToComponent');
const renderer = require('react-test-renderer');

class Hello extends React.Component {
  render() {
    return React.createElement('div', null, `Hello ${this.props.value}`);
  }
}

const factories = {
  'code': (props, children) =>
    React.createElement('pre', {id: 'codeFactory'}, children),
};

describe('mdxToComponent', () => {
  test('render valid', () => {
    return getFilesWithMdx('test/pages/*.md').then(pages => {
      expect(pages.length).toBe(1);
      const Component = mdxToComponent(pages[0].content);
      const res = renderer.create(Component({
        factories,
        value: 'React Value',
        Hello,
      }));
      expect(res.toJSON()).toMatchSnapshot();
    });
  });

  test('render without prop', () => {
    const content = `<Hello value={value} />`
    const Component = mdxToComponent(content);
    const res = renderer.create(Component({
      factories,
      value: 'React Value',
      Hello,
    }));
    expect(res.toJSON()).toMatchSnapshot();
  });
});
