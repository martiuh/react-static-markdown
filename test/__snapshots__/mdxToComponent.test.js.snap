// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mdxToComponent 1`] = `
<div>
  <h1>
    Hey 
    React Value
    !
  </h1>
  <h1
    id="Header"
  >
    Header
  </h1>
  <p>
    Sample text
  </p>
  <pre
    id="codeFactory"
  >
    Sample code block

  </pre>
  <div>
    Hello React Value
  </div>
</div>
`;
